### PR TITLE
Use new GraphQL clients in shopify-api

### PR DESCRIPTION
**Pull request by paulomarg**
### WHY are these changes introduced?

Now that `@shopify/storefront-api-client` and `@shopify/admin-api-client` are released, we want to start using them in this package.

### WHAT is this pull request doing?

Underpinning the existing clients with the new ones. As part of this change, I also did a bit of refactoring on the tests and rearranged the files inside `lib/clients` to match the exported object more closely.

Since this is already a big change (mostly due to tests moving around), I'm setting up an update of the REST client in a different PR.

## Type of change

- [x] Minor: New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have used `yarn changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` file manually)
- [x] I have added/updated tests for this change
- [x] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

File: packages/shopify-api/docs/migrating-to-v9.md
Comment (paulomarg) [reply packages/shopify-api/docs/migrating-to-v9.md:115]: No, I should fix this :)

File: packages/graphql-client/src/graphql-client/types.ts
Comment (paulomarg) [inline packages/graphql-client/src/graphql-client/types.ts:37]: @melissaluu nit-y question: do you think these types should be in the API utilities instead of in graphql-client?

File: packages/shopify-api/lib/__test-helpers__/to-match-made-http-request.ts
Comment (byrichardpowell) [inline packages/shopify-api/lib/__test-helpers__/to-match-made-http-request.ts:77]: Could this error message be improved? E.g: What is the "right" HTTP requests? 

In a test helper like this I'd expect to see `expected` and `saw`

File: packages/shopify-api/lib/index.ts
Comment (byrichardpowell) [inline packages/shopify-api/lib/index.ts:54]: Why is this change a future flag? I assume it's not a breaking change, is it just that we want to eventually remove some code?

File: packages/graphql-client/src/graphql-client/types.ts
Comment (paulomarg) [reply packages/graphql-client/src/graphql-client/types.ts:37]: Ok, thanks!

File: packages/shopify-api/docs/migrating-to-v9.md
Comment (paulomarg) [reply packages/shopify-api/docs/migrating-to-v9.md:16]: Whoops, you're absolutely right. I forgot to fix this back when I reverted.

File: .changeset/fifty-buckets-love.md
Comment (paulomarg) [reply .changeset/fifty-buckets-love.md:8]: I think the best thing here would be to add a documentation page that goes over that step by step. That page can include a video / gif of the experience, plus instructions on setting it up. Maybe we can make that a follow up and ensure we link there from the changelog / migration guide?

File: packages/shopify-api/lib/index.ts
Comment (paulomarg) [reply packages/shopify-api/lib/index.ts:54]: This is actually not a future flag, but there were some types here that were a bit messy, and I'd already had to spend some time fixing them before rolling back. I figured we could keep the fixes around - this is mostly for the benefit of improving `loadRestResources`.

File: packages/shopify-api/lib/__test-helpers__/to-match-made-http-request.ts
Comment (paulomarg) [reply packages/shopify-api/lib/__test-helpers__/to-match-made-http-request.ts:77]: This is actually a non-error - `pass` is set to `true` so this message will never render. I'll change it to a noop to prevent confusion.

File: packages/shopify-api/docs/guides/graphql-types.md
Comment (byrichardpowell) [inline packages/shopify-api/docs/guides/graphql-types.md:20]: Minor, but worth adding npm & pnpm versions?

File: packages/shopify-api/docs/guides/graphql-types.md
Comment (byrichardpowell) [inline packages/shopify-api/docs/guides/graphql-types.md:28]: Can we provide a sample one here? Might not be practical since this package is kinda of generic?

**Review by byrichardpowell — APPROVED** [review]
3 small comments, but approving so you don't need another review.

Great work.

File: packages/shopify-api/docs/guides/graphql-types.md
Comment (paulomarg) [reply packages/shopify-api/docs/guides/graphql-types.md:70]: This is to encourage folks to run it once without watch, because codegen doesn't support caching the schema (which kind of baffles me), so we have to do it by saving it to a file.

On runs subsequent to the first one, the schema is pulled from the file and much much quicker. So this is my little hacky way of steering folks toward the more efficient way to do it.

**Pull request merged**
